src/transforms/geoip.rs

```diff
@@ -165,113 +188,141 @@ struct City<'a> {
                         }
+                    }
 
-                        // last subdivision is most specific per https://github.com/maxmind/GeoIP2-java/blob/39385c6ce645374039450f57208b886cf87ade47/src/main/java/com/maxmind/geoip2/model/AbstractCityResponse.java#L96-L107
-                        if let Some(subdivision) = data.subdivisions.as_ref().and_then(|s| s.last())
+                    // last subdivision is most specific per https://github.com/maxmind/GeoIP2-java/blob/39385c6ce645374039450f57208b886cf87ade47/src/main/java/com/maxmind/geoip2/model/AbstractCityResponse.java#L96-L107
+                    if let Some(subdivision) = data.subdivisions.as_ref().and_then(|s| s.last()) {
+                        if let Some(name) = subdivision
+                            .names
+                            .as_ref()
+                            .and_then(|names| names.get(&*self.locale))
                         {
-                            if let Some(name) = subdivision
-                                .names
-                                .as_ref()
-                                .and_then(|names| names.get(&*self.locale))
-                            {
-                                city.region_name = name;
-                            }
-
-                            if let Some(iso_code) = subdivision.iso_code {
-                                city.region_code = iso_code
-                            }
+                            city.region_name = name;
                         }
 
-                        if let Some(postal_code) = data.postal.and_then(|p| p.code) {
-                            city.postal_code = postal_code;
+                        if let Some(iso_code) = subdivision.iso_code {
+                            city.region_code = iso_code
                         }
                     }
+
+                    if let Some(postal_code) = data.postal.and_then(|p| p.code) {
+                        city.postal_code = postal_code;
+                    }
                 }
-                Err(error) => {
-                    emit!(GeoipIpAddressParseError {
-                        error,
-                        address: ipaddress
-                    });
-                }
+
+                Some(Value::City(city))
             }
-        } else {
-            emit!(ParserMissingFieldError {
-                field: &self.source
-            });
-        };
+            (_, Some(Err((error, ipaddress)))) => {
+                emit!(GeoipIpAddressParseError {
+                    error,
+                    address: ipaddress
+                });
+
+                None
+            }
+            (_, None) => {
+                emit!(ParserMissingFieldError {
+                    field: &self.source
+                });
 
-        let json_value = if self.has_isp_db() {
-            serde_json::to_value(isp)
-        } else {
-            serde_json::to_value(city)
+                None
+            }
```

Comment:
Hmmm, I think this is actually a breaking change, as all of the fields for the given MaxMind database (i.e. the city-specific fields if using the City database, etc) would still be added to the target field, albeit empty, if the IP wasn't parseable or if the source field was missing.

Now the target field will just be `null`.

@jszwedko What do you think? I don't have a strong opinion on `null` overall vs having all the target fields being present but empty.